analyzer/src/main/kotlin/Analyzer.kt

```diff
@@ -58,7 +58,7 @@ class Analyzer(private val config: AnalyzerConfiguration) {
     fun analyze(
         absoluteProjectPath: File,
         packageManagers: List<PackageManagerFactory> = PackageManager.ALL,
-        packageCurationsFile: File? = null,
+        curationProviders: List<PackageCurationProvider> = emptyList(),
```

Comment:
From this classes perspective having an interface which `gives all curations for a package in order` is in my opinion exactly what it needs. So I cannot see what value it brings to make this a list while I think testing becomes a bit harder.

I propose to instead implement a new class e.g. `CompoundPackageCurationProvider` which accepts an arbitrary amount of `CurationProvider` in the constructor and implement just that same interface.

What do you think ?

Reply:
I had the same thought initially, but then couldn't really see a benefit of introducing yet another class that is in fact not a provider on its own, but just a wrapper around existing providers, so I chose this implementation which IMO is more straight-forward and easier to understand. But since you mention it, I'll probably go for the more flexible / OO-like aproach anyway.

Reply:
:-) Thanks!